libpromises/loading.c

```diff
@@ -253,6 +255,44 @@ static void ShowContext(EvalContext *ctx)
     SeqDestroy(soft_contexts);
 }
 
+static void RenameMainBundle(EvalContext *ctx, GenericAgentConfig *config, Policy *policy)
+{
+    const char *const entry_point = EvalContextGetEntryPoint(ctx);
+    Seq *bundles = policy->bundles;
+    int length = SeqLength(bundles);
+    int main_counter = 0;
+    bool removed = false;
+    for (int i = 0; i < length; ++i)
+    {
+        Bundle *const bundle = SeqAt(bundles, i);
+        const char *name = bundle->name;
+        if (StringSafeEqual(name, "__main__"))
+        {
+            const char *const path = bundle->source_path;
+            char *abspath = GetAbsolutePath(path);
+            if (StringSafeEqual(abspath, entry_point))
```

Comment:
Do we have to worry about symlinks here?

Reply:
Yes, I thought about this too, I'll see if we have a realpath function.